From 4cff1f034e26c10ed78d8976b9c21f0437d8009c Mon Sep 17 00:00:00 2001
From: Subrata Chatterjee <schatterjee@habana.ai>
Date: Thu, 11 Mar 2021 08:39:44 +0200
Subject: [PATCH] [SW-34719] gaudi: ras: fix bug in err_sts

Fixed following issues in the err_sts i2c slave
direct command implementation.
     - stored the register value before clear
     - removed explicit initialize to 0 for glabal variable
       as it showed checkpatch error(introduced in already merged patch
       https://gerrit.habana-labs.com/#/c/81265/)

Change-Id: I72c780f25225216432b19f0c055de79915df9dc9
Signed-off-by: Subrata Chatterjee <schatterjee@habana.ai>
---

diff --git a/common/module_i2c_slave.c b/common/module_i2c_slave.c
index 4162534..3c9bca4 100644
--- a/common/module_i2c_slave.c
+++ b/common/module_i2c_slave.c
@@ -241,22 +241,26 @@
 #define ERR_STS_REG_MASK_HI	0xFF00
 u8 i2cslave_error_status_cmd(u8 cmd)
 {
+	u8 val = 0;
+
 	switch (cmd) {
 	case HL_ERROR_STATUS_ALERT_CAUSE_1:
+		/* return the stored low byte */
+		val = err_sts_reg_val & ERR_STS_REG_MASK_LOW;
 		/* clear the low byte as reg is RC(read and clear) */
 		err_sts_reg_val &= ERR_STS_REG_MASK_HI;
-		/* return the stored low byte */
-		return (err_sts_reg_val & ERR_STS_REG_MASK_LOW);
+		break;
 	case HL_ERROR_STATUS_ALERT_CAUSE_2:
-		/* clear the high byte as reg is RC(read and clear) */
-		 err_sts_reg_val &= ERR_STS_REG_MASK_LOW;
 		/* return the stored high byte */
-		return ((err_sts_reg_val & ERR_STS_REG_MASK_HI) >> 8);
+		val = (err_sts_reg_val & ERR_STS_REG_MASK_HI) >> 8;
+		/* clear the high byte as reg is RC(read and clear) */
+		err_sts_reg_val &= ERR_STS_REG_MASK_LOW;
+		break;
 	default:
 		break;
 	}
 
-	return 0;
+	return val;
 }
 
 static inline u8 *i2cslave_consume_hw(u8 *data, u8 cnt)
diff --git a/preboot/preboot.c b/preboot/preboot.c
index ba5521c..e299740 100644
--- a/preboot/preboot.c
+++ b/preboot/preboot.c
@@ -250,7 +250,7 @@
 }
 
 u32 device_id_vendor_id;
-u32 err_sts_reg_val = 0;
+u32 err_sts_reg_val;
 
 static inline void preboot_retrieve_device_id_from_pcie(void)
 {
